src/time_object.cpp

```diff
@@ -244,6 +266,76 @@ Value TimeObject::year(Env *) const {
     return Value::integer(m_time.tm_year + 1900);
 }
 
+nat_int_t TimeObject::normalize_field(Env *env, Value val) {
+    if (!val->is_integer()) {
+        if (val->respond_to(env, "to_i"_s)) {
+            val = val->send(env, "to_i"_s);
+        } else if (val->respond_to(env, "to_int"_s)) {
+            val = val->send(env, "to_int"_s);
+        }
+    }
+    val->assert_type(env, Object::Type::Integer, "Integer");
+    return val->as_integer()->to_nat_int_t();
+}
+
+nat_int_t TimeObject::normalize_field(Env *env, Value val, nat_int_t minval, nat_int_t maxval) {
+    if (val->is_nil()) return minval;
+    auto ival = normalize_field(env, val);
+    if (ival < minval || ival > maxval) {
+        env->raise("ArgumentError", "argument out of range");
+    }
+    return ival;
+}
+
+nat_int_t TimeObject::normalize_month(Env *env, Value val) {
+    if (val->is_nil()) return 0;
+    if (!val->is_integer()) {
+        if (!val->is_string() && val->respond_to(env, "to_str"_s)) {
+            val = val->send(env, "to_str"_s)->as_string();
+        }
+        if (val->is_string()) {
+            auto monstr = val->as_string()->downcase(env, nullptr, nullptr)->as_string()->string();
+            if (isdigit(monstr[0])) {
```

Comment:
If we use `monstr->at(0)` we get bounds checking for free. Otherwise, this could be an out-of-bounds read? (Unless I'm missing a check somewhere.)

Reply:
Oh that's a good point. I'm not sure any of the existing specs had the empty strings which we'd want to make sure to guard against.  I'll look into improving this here.

Reply:
This won't read out of bounds: if it's an empty string, this would read the null terminator, which is not a digit, which simply means this yields `false`.